experimental/SetPartitions/src/Util.jl

```diff
@@ -123,3 +123,47 @@ function _add_partition_top_bottom(vector::Vector{Dict{Int, Set{T}}}, p::T) wher
 
     return vector
 end
+
+"""
+    simplify_operation(partition_sum::Vector{Tuple{S, T}}) where { S <: AbstractPartition, T <: RingElement }
+
+Simplify the vector representation of a `LinearPartition` in terms of distributivity.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d")
+(Univariate polynomial ring in d over QQ, d)
+
+julia> simplify_operation([(set_partition([1, 1], [1, 1]), S(10)), (set_partition([1, 1], [1, 1]), 4*d)])
+Dict{SetPartition, QQPolyRingElem} with 1 entry:
+  SetPartition([1, 1], [1, 1]) => 4*d + 10
+```
+"""
+function simplify_operation(partition_sum::Vector{Tuple{S, T}}) where { S <: AbstractPartition, T <: RingElement }
+
+    partitions = Dict{S, T}()
+
+    for (i1, i2) in partition_sum
+        if iszero(i2)
+            continue
+        end
+        partitions[i1] = get(partitions, i1, 0) + i2
+    end
+    
+    return [(s, t) for (s, t) in partitions]
```

Comment:
```suggestion
    return [(s, t) for (s, t) in partitions if !iszero(t)]
```
You need to filter out zeros in the end again. Because some partition may be contained twice in the list, where the coefficients are additive inverse to each other